File: 2Arrays and Strings/Code/JavaScript/2-2SortedArrayTargetSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {boolean}
 */
var checkForTarget = function(nums, target) {
    let left = 0;
    let right = nums.length - 1;
    
    while (left < right) {
        // curr is the current sum
        let curr = nums[left] + nums[right];
        if (curr == target) {
            return true;
        }
        
        if (curr > target) {
            right--;
        } else {
            left++;
        }
    }
    
    return false;
}

console.log(checkForTarget([1, 2, 4, 6, 8, 9, 14, 15], 13)); // prints true

console.log(checkForTarget([3, 6, 21, 23, 25], 27)); // prints true

console.log(checkForTarget([1, 4, 7, 12, 16], 15)); // prints false

// Compilation steps:
// node 2-2SortedArrayTargetSum.js

// Time-complexity : O(n)
// Space complexity: O(1)

File: 2Arrays and Strings/Code/Python/2-2SortedArrayTargetSum.py
def check_for_target(nums, target):
    left = 0
    right = len(nums) - 1

    while left < right:
        # curr is the current sum
        curr = nums[left] + nums[right]
        if curr == target:
            return True
        if curr > target:
            right -= 1
        else:
            left += 1
    
    return False

# test-cases
print(check_for_target([1, 2, 4, 6, 8, 9, 14, 15], 13)) # true

print(check_for_target([3, 6, 21, 23, 25], 27)) # true

print(check_for_target([1, 4, 7, 12, 16], 15)) # false
